core/src/main/java/com/es/core/model/exceptions/PhoneNotFoundException.java

```diff
@@ -0,0 +1,14 @@
+package com.es.core.model.exceptions;
+
+import lombok.Data;
+
+@Data
+public class PhoneNotFoundException extends RuntimeException {
+    public int errorCode;
+    public String errorMessage;
+
+    public PhoneNotFoundException() {
+        this.errorCode = 404;
```

Comment:
Is HTTP code really a responsibility of that exception? 🤔 
It can be the case for example when you try to implement some general exception for DB operations, or maybe when you try to extend such exception. But here you specific enough about what case handled and I think it's always going to be 404 in that case. Generally it's exception handler responsibility to build (or to initiate build) of HTTP response, so I would expect HTTP code to be specified there. At least in this case